docs/DeveloperGuide.md

```diff
@@ -1,38 +1,181 @@
 # Developer Guide
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```

Comment:
Maybe before explaining the design and implementation you can add in a brief summary of each component's functions and responsibilities for easier understanding of the system